tests/test_deregister_task_definition.py

```diff
@@ -0,0 +1,232 @@
+                ),
+            )
+
+            ecs_describe_services_patch.assert_called()
+            ecs_deregister_patch.assert_has_calls(
+                [
+                    call(
+                        taskDefinition=local_task_definition_1,
+                    ),
+                    call(taskDefinition=production_task_definition_1),
+                    call(taskDefinition=preflight_task_definition_1),
+                ],
+                any_order=True,
+            )
+
+    def test_deregister_task_definition_failed_deployment(self):
+        local_task_definition_1 = Mock(return_value="local_exec_arn_1")
+        production_task_definition_1 = Mock(return_value="production_arn_1")
+        preflight_task_definition_1 = Mock(return_value="preflight_arn_1")
+        local_task_definition_2 = Mock(return_value="local_exec_arn_2")
+        production_task_definition_2 = Mock(return_value="production_arn_2")
+        preflight_task_definition_2 = Mock(return_value="preflight_arn_2")
+
+        with (
+            self.subTest("Failed deployment, roll backed to previous"),
+            patch.object(
+                self.ecs_client,
+                "describe_services",
+                return_value=self.describe_service_return_value(
+                    "PRIMARY",
+                    arn=production_task_definition_1,
+                ),
+            ) as ecs_describe_services_patch,
+            patch.object(self.ecs_client, "deregister_task_definition") as ecs_deregister_patch,
+            self.assertRaises(SystemExit),
+        ):
+            deregister_task_definition(
+                ecs_client=self.ecs_client,
+                service="",
+                cluster="",
+                local_task_definition_output=CreateTaskDefinitionOutput(
+                    previous_task_definition_arn=local_task_definition_1,
+                    latest_task_definition_arn=local_task_definition_2,
+                ),
+                production_task_definition_output=CreateTaskDefinitionOutput(
+                    previous_task_definition_arn=production_task_definition_1,
+                    latest_task_definition_arn=production_task_definition_2,
+                ),
+                preflight_task_definition_output=CreateTaskDefinitionOutput(
+                    previous_task_definition_arn=preflight_task_definition_1,
+                    latest_task_definition_arn=preflight_task_definition_2,
+                ),
+            )
+
+            ecs_describe_services_patch.assert_called()
+            ecs_deregister_patch.assert_has_calls(
+                [
+                    call(
+                        taskDefinition=local_task_definition_2,
+                    ),
+                    call(taskDefinition=production_task_definition_2),
+                    call(taskDefinition=preflight_task_definition_2),
+                ],
+                any_order=True,
```

Comment:
Why do you need any order (here and in other places)? Can it change? I'm not sure I understand this correctly.